### PR TITLE
fix: Fix copying metadata during lowering

### PR DESCRIPTION
**Pull request by peri044**
# Description

Fixes https://github.com/pytorch/TensorRT/issues/3221

## Type of change

Please delete options that are not relevant and/or add your own.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- This change requires a documentation update

# Checklist:

- [ ] My code follows the style guidelines of this project (You can use the linters)
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests to verify my fix or my feature
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

**Review by narendasan — APPROVED** [review]
Would an alternative be to rerun the fake mode after lowering?

**Comment by peri044** [comment]
> Would an alternative be to rerun the fake mode after lowering?

Yes, that would be an alternative. However, that requires us to maintain user provided inputs and dynamic shape range during tracing (provided at `torch.export.export()`). These inputs should be used via `ShapeTensorProp` or `FakeTensorProp`. Also, using this can modify the metadata for other ops in the graph which can get complicated.
